test/e2e/apimachinery/crd_validation_rules.go

```diff
@@ -0,0 +1,194 @@
+	framework.ExpectNoError(err, "loading config")
+
+	dynamicClient, err := dynamic.NewForConfig(config)
+	framework.ExpectNoError(err, "initializing dynamic client")
+	gvr := schema.GroupVersionResource{
+		Group:    crd.Spec.Group,
+		Version:  crd.Spec.Versions[0].Name,
+		Resource: crd.Spec.Names.Plural,
+	}
+
+	ginkgo.It("MUST fail creation of a custom resource definition with x-kubernetes-validator rules containing invalid syntax", func() {
+		e2eskipper.SkipUnlessFeatureGateEnabled(features.CustomResourceValidationExpressions)
+		_, err = setupCRD(f, schemaWithIllegalValidationExpression, "common-group", "v1")
+		framework.ExpectError(err, "x-kubernetes-validator rules containing invalid syntax")
+		expectedErrMsg := "compilation failed"
+		if !strings.Contains(err.Error(), expectedErrMsg) {
+			framework.Failf("expect error contains %q, got %q", expectedErrMsg, err.Error())
+		}
+	})
+	ginkgo.It("MUST NOT fail validation for create and update of a custom resource that satisfies the x-kubernetes-validator rules", func() {
+		// features.CustomResourceValidationExpressions can be enabled or disabled for this test
+		// since validation is skipped when the feature flag is disabled
+		crClient := dynamicClient.Resource(gvr)
+
+		name1 := names.SimpleNameGenerator.GenerateName("cr-1")
+		cr, err := crClient.Create(context.TODO(), &unstructured.Unstructured{Object: map[string]interface{}{
+			"apiVersion": gvr.Group + "/" + gvr.Version,
+			"kind":       crd.Spec.Names.Kind,
+			"metadata": map[string]interface{}{
+				"name": name1,
+			},
+			"spec": map[string]interface{}{
+				"x": int64(1),
+				"y": int64(0),
+			},
+		}}, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "validation rules satisfied")
+
+		cr.Object["status"] = map[string]interface{}{
+			"health": "ok",
+		}
+		_, err = crClient.UpdateStatus(context.TODO(), cr, metav1.UpdateOptions{})
+		framework.ExpectNoError(err, "status validation rules satisfied")
+	})
+	ginkgo.It("MUST fail validation for create of a custom resource that does not satisfy the x-kubernetes-validator rules", func() {
+		e2eskipper.SkipUnlessFeatureGateEnabled(features.CustomResourceValidationExpressions)
+		crClient := dynamicClient.Resource(gvr)
+
+		name1 := names.SimpleNameGenerator.GenerateName("cr-1")
+		_, err = crClient.Create(context.TODO(), &unstructured.Unstructured{Object: map[string]interface{}{
+			"apiVersion": gvr.Group + "/" + gvr.Version,
+			"kind":       crd.Spec.Names.Kind,
+			"metadata": map[string]interface{}{
+				"name": name1,
+			},
+			"spec": map[string]interface{}{
+				"x": int64(0),
+				"y": int64(0),
+			},
+		}}, metav1.CreateOptions{})
+		framework.ExpectError(err, "validation rules not satisfied")
+		expectedErrMsg := "rule failed"
+		if !strings.Contains(err.Error(), expectedErrMsg) {
+			framework.Failf("expect error contains %q, got %q", expectedErrMsg, err.Error())
+		}
+	})
+})
+
+var schemaWithValidationExpression = []byte(`
```

Comment:
var or constant?

if var better inside the SIGDescribe block

Reply:
I've reformatted the test a bit to be more ginkgo idiomatic, including moving the json schema into the describe block.